x-pack/plugins/secops/public/pages/hosts/index.tsx

```diff
@@ -14,57 +14,68 @@ import { EmptyPage } from '../../components/empty_page';
-          <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
-            {({ kpiEventType, loading }) => (
-              <TypesBar
-                loading={loading}
-                data={
-                  kpiEventType!.map((i: KpiItem) => ({
-                    x: i.count,
-                    y: i.value,
-                  })) as HorizontalBarChartData[]
-                }
-              />
-            )}
-          </EventsQuery>
-          <HostsQuery sourceId="default" startDate={startDate} endDate={endDate} cursor={null}>
-            {({ hosts, totalCount, loading, pageInfo, loadMore }) => (
-              <HostsTable
-                loading={loading}
-                data={hosts}
-                totalCount={totalCount}
-                hasNextPage={getOr(false, 'hasNextPage', pageInfo)!}
-                nextCursor={getOr(null, 'endCursor.value', pageInfo)!}
-                loadMore={loadMore}
-              />
-            )}
-          </HostsQuery>
-          <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
-            {({ events, loading }) => (
-              <EventsTable
-                data={events!}
-                loading={loading}
-                startDate={startDate}
-                endDate={endDate}
-              />
-            )}
-          </EventsQuery>
-        </>
+        <GlobalTime>
+          {({ poll, to, from, setQuery }) => (
+            <>
+              <EventsQuery sourceId="default" startDate={to} endDate={from}>
+                {({ kpiEventType, loading }) => (
+                  <TypesBar
+                    loading={loading}
+                    data={
+                      kpiEventType!.map((i: KpiItem) => ({
+                        x: i.count,
+                        y: i.value,
+                      })) as HorizontalBarChartData[]
+                    }
+                  />
+                )}
+              </EventsQuery>
+              <HostsQuery sourceId="default" startDate={to} endDate={from} poll={poll}>
+                {({ hosts, totalCount, loading, pageInfo, loadMore, id, refetch }) => (
+                  <HostsTableManage
+                    id={id}
+                    refetch={refetch}
+                    setQuery={setQuery}
+                    loading={loading}
+                    data={hosts}
+                    totalCount={totalCount}
+                    hasNextPage={getOr(false, 'hasNextPage', pageInfo)!}
+                    nextCursor={getOr(null, 'endCursor.value', pageInfo)!}
+                    loadMore={loadMore}
+                  />
+                )}
+              </HostsQuery>
+              <EventsQuery sourceId="default" startDate={to} endDate={from}>
+                {({ events, loading, id, refetch }) => (
+                  <EventsTableManage
+                    id={id}
+                    refetch={refetch}
+                    setQuery={setQuery}
+                    data={events!}
```

Comment:
Kind of the same line of thinking of removing the `!` for null/undefined values when possible for better type safety here. I removed this and it works fine without the `!`. The `!` can cause runtime issues and should be used sparingly.

I would remove this one.

Reply:
I am getting an error when I removed it

Reply:
I did too. Ignore the advice. My apologies.